src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPSavableTab.cpp

```diff
@@ -49,6 +49,8 @@ QVariant PCRPrimerDesignForDNAAssemblyOPSavableTab::getChildValue(const QString&
         return QVariant::fromValue<ResultTableData>(productTable->getPCRPrimerProductTableData());
     } else if (userPrimerLineEdit != nullptr) {
         return QVariant::fromValue<UserPimerLineEditResult>(userPrimerLineEdit->getData());
+    } else if (childId.endsWith("AreaSelectManually")) {
```

Comment:
Argh, don't think, that I am too picky or trying to pick on you, but it's not good again, If someone someday for some reason change the widget IDs, this code will stop working and it'll be very difficult to find that it's because the part of ID which's been changed is used here. This is the case when the best solution is already suggested and you don't need to invent something new. Just do as it's done several lines above - use `qobject_cast`